server/pkg/proxystore/proxystore.go

```diff
@@ -61,16 +59,16 @@ func New() *Store {
 		c:    sync.NewCond(&sync.Mutex{}),
 		tree: btree.New(2),
 		sync: map[Set]bool{},
-		pb:   proto.NewBuffer(make([]byte, 0)),
 	}
 }
 
 func (s *Store) hashOf(m proto.Message) (h uint64) {
-	if err := s.pb.Marshal(m); err != nil {
+	message, err := proto.Marshal(m)
+	if err != nil {
 		panic(err) // should not happen
 	}
-	h = xxhash.Sum64(s.pb.Bytes())
-	s.pb.Reset()
+	h = xxhash.Sum64(message)
+	proto.Reset(m)
```

Comment:
the original `Reset` was resetting the *buffer*, we don't want to reset the given message ;-) With `proto.Marshal` you have a new []byte every time, so no need to reset anything.